### ISSUE TITLE
[CLI]: `run.history()` missing nearly all data when logging with PyTorch Lightning 

### ISSUE THREAD

**Issue by jlehrer1** [wandb/wandb#3680]
### Describe the bug

When logging metrics to Wandb.ai with PyTorch Lightning's `WandbLogger` object and then fetching run history with `wandb.Api().run("...")`, nearly all data is missing from the returned `DataFrame`.

```python 
run = api.run("jlehrer1/Retina Model/1df1l0u4")

df = run.history()
df.shape
>>> (503, 36)

df.isna().sum()
>>>
trainer/global_step            0
_step                          0
_runtime                       0
lr-Adam                      496
_timestamp                     0
train_loss_step              474
epoch                        471
val_loss_step                 82
val_median_f1                496
val_auroc                    496
val_per_class_recall         496
val_confusion_matrix         496
val_f1                       496
val_total_accuracy           496
val_recall                   496
val_balanced_accuracy        496
val_weighted_accuracy        496
val_loss_epoch               496
val_per_class_f1             496
val_precision                496
val_per_class_precision      496
val_specificity              496
train_precision              497
train_f1                     497
train_auroc                  497
train_specificity            497
train_loss_epoch             497
train_confusion_matrix       497
train_weighted_accuracy      497
train_balanced_accuracy      497
train_total_accuracy         497
train_per_class_recall       497
train_per_class_precision    497
train_recall                 497
train_per_class_f1           497
test_loss_step               445
dtype: int64
```

Interestingly, I am logging train metrics and validation metrics in precisely the same way. In fact the exact same logging method is being called in my PyTorch Lightning datamodule, defined as follows:

```
    def _compute_metrics(self, 
        y_hat: torch.Tensor, 
        y: torch.Tensor, 
        tag: str,
        on_epoch=True, 
        on_step=False,
    ):
        for name, metric in self.metrics.items():
            val = metric(y_hat, y)
            self.log(
                f"{tag}_{name}", 
                val, 
                on_epoch=on_epoch, 
                on_step=on_step,
                logger=True,
            )
```

For context, `self.metrics` simply contains a list of methods that take in logits / label tensors and returns particular metrics I'd like to track. 

I'm guessing there are other people using Wandb and PyTorch Lightning successfully, so any pointers on where I'm going wrong would be great!

### Environment

WandB version: `0.12.11`

OS: MacOS Monterey (Apple Silicon chip)

Python version: `Python 3.9.7`

Versions of relevant libraries: `pytorch-lightning==1.5.4`

**Comment by vanpelt** [comment]
`.history` returns samples from the raw history.  If you want the full resolution history you should use [.scan_history](https://docs.wandb.ai/guides/track/public-api-guide#get-unsampled-metric-data)

**Comment by MBakirWB** [comment]
Hi @jlehrer1,

I will be closing this out. Please do reach out again if you require additional support or assistance.

**Comment by ArthurConmy** [comment]
IMO this is extremely misleading and `.history()` should at very least throw a warning when some data is not covered. This appeared as a transient error in my project and was frustrating.
